src/solvers/smt2/smt2_command.h

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************\
+
+Module: SMT2 solver
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// SMT2 commands
+
+#ifndef CPROVER_SOLVERS_SMT2_SMT2_COMMAND_H
+#define CPROVER_SOLVERS_SMT2_SMT2_COMMAND_H
+
+#include "smt2_ast.h"
+
+class smt2_command_or_commentt
+{
+public:
+  virtual void print(std::ostream &stream) const = 0;
+  virtual ~smt2_command_or_commentt() = default;
+};
+
+inline std::ostream &
+operator<<(std::ostream &stream, const smt2_command_or_commentt &command)
+{
+  command.print(stream);
+  return stream;
+}
+
+class smt2_commandt final : public smt2_command_or_commentt
```

Comment:
Reference the SMT-LIB standard http://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.0-r10.12.21.pdf